src/test/java/com/iexec/sms/secret/web2/Web2SecretsServiceTests.java

```diff
@@ -101,69 +99,89 @@ void shouldGetEmptyResultIfSecretNotPresent() {
 
     @Test
     void shouldNotAddSecretIfPresent() {
-        Web2Secrets web2Secrets = new Web2Secrets(ownerAddress);
-        web2Secrets.getSecrets().add(new Secret(secretAddress, encryptedSecretValue));
+        Web2Secrets web2Secrets = new Web2Secrets(ownerAddress, List.of(new Secret(secretAddress, encryptedSecretValue, true)));
         when(web2SecretsRepository.findWeb2SecretsByOwnerAddress(ownerAddress)).thenReturn(Optional.of(web2Secrets));
-        assertThat(web2SecretsService.addSecret(ownerAddress, secretAddress, plainSecretValue)).isFalse();
-        verifyNoInteractions(encryptionService);
```

Comment:
I preferred the previous implementation
Why should we need to encrypt secret before checking if it already exists ?

Reply:
We want to avoid double checks. As `Web2Secrets` should manage its own `Secret`s, I'd prefer the check to be there.
IMHO, it'll be solved in our next PR on Web2 & Web3 architecture sharing.